openshift/machine-config-operator
# OCPBUGS-11146: update cloud provider flags on vSphere for storage options

**- What I did**

* modify the render pipeline to include the storage operator object
* update the cloud provider flag to check the storage options when using vSphere
* only use external cloud provider flag on vSphere when CSI storage is enabled

**- How to verify it**

there are 3 main conditions to verify with this change:

1. starting with a 4.12 vSphere cluster using in-tree storage, upgrade to 4.13 should preserve the `--cloud-provider=vsphere` flag on the kubelet while using `--cloud-provider=external` on the kube-apiserver and kube-controller-manager.
2. starting with a 4.12 vSphere cluster using out-of-tree storage (CSI), upgrade to 4.13 should upgrade all cloud provider flags to `--cloud-provider=external`.
3. installing new clusters on 4.13 should use all cloud provider flags as `--cloud-provider=external`.

**- Description for the changelog**

Update external cloud provider check for vSphere to observe the CSI migration status and properly configure the cluster for in-tree storage and out-of-tree cloud providers.
